GitUI/CommandsDialogs/UserScriptContextMenuExtensions.cs

```diff
@@ -45,7 +46,8 @@ public static void AddUserScripts(this ContextMenuStrip contextMenu, ToolStripMe
                 item.Click += (s, e) =>
                 {
                     int scriptId = script.HotkeyCommandIdentifier;
-                    scriptInvoker(scriptId);
+                    var scriptsRunner = serviceProvider.GetRequiredService<IScriptsRunner>();
+                    scriptsRunner.RunScript(scriptId, hostControl.FindForm() as GitModuleForm, hostControl);
```

Comment:
I actually like the idea of `Control IScriptHostControl.AsControl()`, since it also allows `IScriptHostControl` to not itself be a control, and instead be a separate implementation. Which, in retrospect, I need for the file scripts PR. However, this can be farther optimized to `GitModuleForm GetGitModuleForm()`, and skip the middleman.

Reply:
P.S. There's a chance here for an even bigger refactor, to remove `THostForm form` from `IScriptsRunner.RunScript`, and instead, pass it inside `IScriptHostControl`. Looking at the usage, the `IGitModuleForm side of it is only used for obtaining an `IGitUICommands`, so it's also a chance to remove `IServiceProvider serviceProvider` from `UserScriptContextMenuExtensions.AddUserScripts`

But I'm not sure if changing that much is a good idea for this one PR.